executor/admin.go

```diff
@@ -0,0 +1,231 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"io"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/distsql"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/terror"
+	"github.com/pingcap/tidb/util/ranger"
+	"github.com/pingcap/tidb/util/types"
+	"github.com/pingcap/tipb/go-tipb"
+)
+
+type idxResult struct {
+	IndexVals [][]types.Datum
+	Handles   []int64
+	LastVal   []types.Datum
+	finished  bool
+}
+
+// CheckIndexExec output the index values which has handle between begin and end.
+type CheckIndexExec struct {
+	baseExecutor
+
+	table    table.Table
+	index    table.Index
+	is       infoschema.InfoSchema
+	startKey []types.Datum
+	begin    int64
+	end      int64
+
+	batch   [][]types.Datum
+	handles []int64
+	idx     int
+	done    bool
+	init    bool
+
+	result   distsql.NewSelectResult
+	partial  distsql.NewPartialResult
+	typesMap map[int64]*types.FieldType
+	cols     []*model.ColumnInfo
+}
+
+// Next implements the Executor Next interface.
+func (e *CheckIndexExec) Next() (Row, error) {
```

Comment:
Do we need to implement NextChunk() interface?

Reply:
release-1.0 doesn't has `NextChunk` interface.